### PR TITLE
Add `@testset` macro to test/linalg/dense.jl

### PR DESCRIPTION
**Pull request by xorJane**
Hello!

I've added the @testset macro to test/linalg/dense.jl and removed the `debug && println(...)` statements formerly included in the file.

When I run `Base.runtests("linalg/dense")` in the `REPL`, I see that all tests are passing; however, only 6240 tests in `dense.jl` register, whereas there were 7212 total on `master`. Should wrapping tests in `@testset` lower the total test count?

Thanks in advance for reviewing!

### CODE REVIEW

File: test/linalg/dense.jl [hunk 1]
```diff
@@ -185,354 +194,372 @@ for elty in (Int32, Int64, Float32, Float64, Complex64, Complex128)
...
+
+        A5  = convert(Matrix{elty}, [1 1 0 1; 0 1 1 0; 0 0 1 1; 1 0 0 1])
+        @test expm(logm(A5)) ≈ A5
+
+        A6  = convert(Matrix{elty}, [-5 2 0 0 ; 1/2 -7 3 0; 0 1/3 -9 4; 0 0 1/4 -11])
+        @test expm(logm(A6)) ≈ A6
+
+        A7  = convert(Matrix{elty}, [1 0 0 1e-8; 0 1 0 0; 0 0 1 0; 0 0 0 1])
+        @test expm(logm(A7)) ≈ A7
+
```
Comment (tkelman) [inline test/linalg/dense.jl]: avoid empty lines as the first or last thing in an indented block

File: test/linalg/dense.jl [hunk 2]
```diff
@@ -185,354 +194,372 @@ for elty in (Int32, Int64, Float32, Float64, Complex64, Complex128)
...
+        @test log(det(eye(N))) ≈ logdet(eye(N))
+    end
 end
 
-let
-    N = 3
-    @test log(det(eye(N))) ≈ logdet(eye(N))
+@testset "issue #2637" begin
+    let
+      a = [1, 2, 3]
```
Comment (tkelman) [inline test/linalg/dense.jl]: This block wasn't indented to the typical multiple of 4 spaces. The `let` is also redundant with the testset now.

**Comment by andreasnoack** [comment]
Thanks for testsetting this file. Looks right to me so I don't understand why the total number of tests would be lower after this change.

**Comment by KristofferC** [comment]
Could be worth to revert https://github.com/JuliaLang/julia/pull/20150 and see if that changes the total number of tests passed.

**Comment by KristofferC** [comment]
I tried it but get 6240 tests even with the commit above reverted.

File: test/linalg/dense.jl (hunk 1 above)
Comment (xorJane) [inline test/linalg/dense.jl]: Great, thanks! Fixed this and another code block where I had done the same thing.

File: test/linalg/dense.jl (hunk 2 above)
Comment (xorJane) [inline test/linalg/dense.jl]: I updated spacing and removed a few redundant  `let` blocks.

**Comment by xorJane** [comment]
Thank you for trying that out!

**Comment by KristofferC** [comment]
I guess the boring way is to not add everything at once and see when the number of tests gets reduced. Perhaps by splitting the file into multiple parts.

File: test/linalg/dense.jl [hunk 3]
```diff
@@ -185,354 +192,358 @@ for elty in (Int32, Int64, Float32, Float64, Complex64, Complex128)
...
-                    @test norm(z, p) == (p < 0 ? 0 : Inf)
-                    @test norm(elty[Inf],p) == Inf
+            @testset "issue #10234" begin
+                if elty <: AbstractFloat || elty <: Complex
+                    z = zeros(elty, 100)
+                    z[1] = -Inf
+                    for p in [-2,-1.5,-1,-0.5,0.5,1,1.5,2,Inf]
+                        @test norm(z, p) == (p < 0 ? 0 : Inf)
+                        @test norm(elty[Inf],p) == Inf
+                    end
```
Comment (xorJane) [inline test/linalg/dense.jl]: Changes to this section of code (lines 335 to 342) are causing the reduction in the number of tests. This code block was formerly wrapped into the `for` loop starting on line 298 (taking `i` from 1 to 10), and I removed it when I added testsets. 

It looks to me like the same tests at lines 340 and 341 were being repeated 10 times (for each type) as they don't appear to have any dependency on `i`, `A`, `B`, or `α`. Does it look like I understand correctly? Thank you!

Comment (andreasnoack) [inline test/linalg/dense.jl]: Yes. That could very well be the explanation.

**Comment by KristofferC** [comment]
Thanks for doing this.

**Pull request merged**